src/main/java/hudson/tasks/junit/CaseResult.java

```diff
@@ -72,6 +72,13 @@
     private transient SuiteResult parent;
 
     private transient ClassResult classResult;
+    /**
+     * An optional archive id to be able to differentiate suites that has been merged.
+     * @see JUnitResultArchiver#archiveId
+     * @see SuiteResult#archiveId
+     */
+    @CheckForNull
+    private final String archiveId;
```

Comment:
this is already available in parent.getArchiveID()

When de-serialized this will no longer be the same string everywhere but a new String - such that if you have 1000 classes you will have 1001 instance of a the same String rather than one instance that is shared - as such it will slow down the loading as well as add more pressure on the GC (lazily loading these does cause issues) - it will also take up (a small amount more diskspace).

Better to just call the parent to get it.
If you feel you really want it locally then make it transient and set it in `freeze()`, `setParentSuiteResult()` `replaceParent()`